tests/Api/Response/ResponseTest.php

```diff
@@ -253,15 +253,98 @@ public function getQueryInfoMessageProvider(): array
                         'position' => 30
                     ]
                 ],
-                'Findologic::Template.queryInfoMessageQuery',
-                [
+                'expectedTranslation' => 'Findologic::Template.queryInfoMessageQuery',
+                'expectedTranslationParams' => [
                     'query' => 'TestQuery',
                     'hits' => 0
                 ]
             ]
         ];
     }
 
+    public function smartDidYouMeanQueryProvider(): array
+    {
+        return [
+            'Smart Did-You-Mean is not triggered if there is an empty query' => [
+                'queryInfoMessageData' => [
+                    'originalQuery' => null,
+                    'didYouMeanQuery' => null,
+                    'currentQuery' => '',
+                    'queryStringType' => null,
+                ],
+                'expectedTranslationKey' => '',
+                'expectedTranslationParams' => [],
+            ],
+            'Smart Did-You-Mean with did you mean query' => [
+                'queryInfoMessageData' => [
+                    'originalQuery' => 'blub',
+                    'didYouMeanQuery' => 'blubbergurken',
+                    'currentQuery' => 'blub',
+                    'queryStringType' => null,
+                ],
+                'expectedTranslationKey' => 'Findologic::Template.didYouMeanQuery',
+                'expectedTranslationParams' => [
+                    'originalQuery' => 'blub',
+                    'alternativeQuery' => 'blubbergurken'
+                ],
+            ],
+            'Smart Did-You-Mean with improved query' => [
+                'queryInfoMessageData' => [
+                    'originalQuery' => 'blub',
+                    'didYouMeanQuery' => null,
+                    'currentQuery' => 'blubbergurken',
+                    'queryStringType' => 'improved',
+                ],
+                'expectedTranslationKey' => 'Findologic::Template.improvedQuery',
+                'expectedTranslationParams' => [
+                    'originalQuery' => 'blub',
+                    'alternativeQuery' => 'blubbergurken'
+                ],
+            ],
+            'Smart Did-You-Mean with corrected query' => [
+                'queryInfoMessageData' => [
+                    'originalQuery' => 'blub',
+                    'didYouMeanQuery' => null,
+                    'currentQuery' => 'blubbergurken',
+                    'queryStringType' => 'corrected',
+                ],
+                'expectedTranslationKey' => 'Findologic::Template.correctedQuery',
+                'expectedTranslationParams' => [
+                    'originalQuery' => 'blub',
+                    'alternativeQuery' => 'blubbergurken'
+                ],
```

Comment:
It would be easier to differentiate between different SDYM cases when "real" terms are used for the queries and not blub and blubbergurken. Think how you can see the SDYM cases in the shop e.g. searching for sesel

Reply:
You can not replicate real cases as everything except the corrected query, is based on the query history, which may lead to such a result.
In this case the given search term is always `blub` and everything that is corrected will be `blubbergurken`. I get that we want to be as real in tests as in the real world, but this could be such a case as well.

Nevertheless I think it is a good idea, so I changed it. Sent query is always `sesel` and corrected/improved/did-you-mean is always `sessel`.

Reply:
This was just an example is there an english equivalent?

Reply:
Sure. Resolved.